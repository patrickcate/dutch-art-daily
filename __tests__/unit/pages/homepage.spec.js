import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueMeta from 'vue-meta'
import Homepage from '@pages/index.vue'

describe('Homepage', () => {
  it('should be a Vue instance', () => {
    const localVue = createLocalVue()
    localVue.use(VueMeta, { keyName: 'head' })

    const wrapper = shallowMount(Homepage, {
      mocks: {
        $homepageRedirect() {},
      },
      localVue,
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('render an empty string', () => {
    const localVue = createLocalVue()
    localVue.use(VueMeta, { keyName: 'head' })

    process.server = true

    const wrapper = shallowMount(Homepage, {
      mocks: {
        $homepageRedirect() {},
      },
      methods: {
        $icon() {
          return true
        },
      },
      localVue,
    })

    expect(wrapper).toMatchSnapshot()
  })
})
