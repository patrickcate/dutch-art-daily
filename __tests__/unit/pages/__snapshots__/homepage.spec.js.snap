// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage render an empty string 1`] = ``;
